src/Elasticsearch/Helper/Iterators/SearchHitIterator.php

```diff
@@ -76,8 +76,8 @@ public function rewind(): void
         }
 
         $this->count = 0;
-        if (isset($current_page['hits']) && isset($current_page['hits']['total'])) {
-            $this->count = $current_page['hits']['total'];
+        if (isset($current_page['hits']) && isset($current_page['hits']['hits'])) {
+            $this->count = count($current_page['hits']['hits']);
```

Comment:
We can use the `$current_page['hits']['total']` value of the result instead of `count()`. We need to check if `total` is a number or an array, since you can switch the behaviour using [rest_total_hits_as_hint](https://www.elastic.co/guide/en/elasticsearch/reference/current/breaking-changes-7.0.html#hits-total-now-object-search-response).
We can use the following code:
```php
if (isset($current_page['hits']) && isset($current_page['hits']['total'])) {
    $this->count = $current_page['hits']['total']['value'] ?? $current_page['hits']['total'];
}
```

Reply:
Looks totally ok for me. I'll update the code.